### PR TITLE
Update push notification to support Android SDK 31+

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

Android 12 included two behavior changes which affected Amplify push notification:
1. [Pending intents mutability](https://developer.android.com/about/versions/12/behavior-changes-12#pending-intent-mutability) - `PendingIntent`s now require mutability flag to be specified
2. [Notification trampoline restrictions](https://developer.android.com/about/versions/12/behavior-changes-12#notification-trampolines) - Services and broadcast receivers can no longer start activities

This PR updates Amplify push notification to address these two behavioral changes:
1. `PendingIntents` now include mutability flag (set to immutable)
2. Activity opening is now built into the notification's content intent. As a result, the broadcast receiver for notifications is bypassed and thus no longer needed.

This PR also cleans up the state of the Android artifacts checked into this repo to remove the `.gradle/` folder but include the gradle wrapper.

#### Issue #, if available

~~https://github.com/aws-amplify/amplify-js/issues/10118~~ May not actually be relevant
https://github.com/aws-amplify/amplify-js/issues/10252

#### Description of how you validated changes
Tested locally with sample Android apps targeting < SDK 31 and >= SDK 31

#### Checklist

- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jimblanc — APPROVED** [review]
LGTM, discussed gradle wrappers offline.

File: packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/modules/RNPushNotificationHelper.java [hunk 1]
```diff
@@ -353,16 +353,11 @@ public void sendToNotificationCentre(Bundle bundle) {
                     // Add "action" for later identifying which button gets pressed.
                     bundle.putString("action", action);
                     actionIntent.putExtra("notification", bundle);
-                    PendingIntent pendingActionIntent = PendingIntent.getBroadcast(context, notificationID, actionIntent,
-                            PendingIntent.FLAG_UPDATE_CURRENT);
+                    PendingIntent pendingActionIntent = PendingIntent.getBroadcast(context, notificationID, actionIntent, getPendingIntentFlags());
```
Comment (Samaritan1011001) [inline packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/modules/RNPushNotificationHelper.java:356]: Should this be `getActivity` too?

**Review by Samaritan1011001 — COMMENTED** [review]
Just two questions.

File: packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/modules/RNPushNotificationHelper.java [hunk 2]
```diff
@@ -78,7 +81,7 @@ private PendingIntent toScheduleNotificationIntent(Bundle bundle) {
         notificationIntent.putExtra(RNPushNotificationPublisher.NOTIFICATION_ID, notificationID);
         notificationIntent.putExtras(bundle);
 
-        return PendingIntent.getBroadcast(context, notificationID, notificationIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+        return PendingIntent.getBroadcast(context, notificationID, notificationIntent, getPendingIntentFlags());
```
[the remark this answers is not in the document]
Comment (cshfang) [reply packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/modules/RNPushNotificationHelper.java:84]: I did not update these as our library does not support scheduled notifications at this time despite including this underlying native code

File: packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/modules/RNPushNotificationHelper.java (hunk 1 above)
Comment (cshfang) [reply packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/modules/RNPushNotificationHelper.java:356]: Ditto here

**Review by Samaritan1011001 — APPROVED** [review]
LGTM.

**Pull request merged**